packages/core/src/sync-store/index.ts

```diff
@@ -528,327 +468,229 @@ export const createSyncStore = ({
-        ])
-        .leftJoin("traces", "traces.id", "event.traceId")
-        .select([
-          "traces.id as trace_id",
-          "traces.type as trace_callType",
-          "traces.from as trace_from",
-          "traces.to as trace_to",
-          "traces.gas as trace_gas",
-          "traces.gasUsed as trace_gasUsed",
-          "traces.input as trace_input",
-          "traces.output as trace_output",
-          "traces.error as trace_error",
-          "traces.revertReason as trace_revertReason",
-          "traces.value as trace_value",
-          "traces.index as trace_index",
-          "traces.subcalls as trace_subcalls",
-        ])
-        .leftJoin(
-          "transactionReceipts",
-          "transactionReceipts.transactionHash",
-          "event.transactionHash",
-        )
-        .select([
-          "transactionReceipts.contractAddress as txr_contractAddress",
-          "transactionReceipts.cumulativeGasUsed as txr_cumulativeGasUsed",
-          "transactionReceipts.effectiveGasPrice as txr_effectiveGasPrice",
-          "transactionReceipts.from as txr_from",
-          "transactionReceipts.gasUsed as txr_gasUsed",
-          "transactionReceipts.logsBloom as txr_logsBloom",
-          "transactionReceipts.status as txr_status",
-          "transactionReceipts.to as txr_to",
-          "transactionReceipts.type as txr_type",
-        ])
-        .where("event.checkpoint", ">", from)
-        .where("event.checkpoint", "<=", to)
-        .orderBy("event.checkpoint", "asc")
-        .orderBy("event.filterIndex", "asc")
-        .$if(limit !== undefined, (qb) => qb.limit(limit!))
-        .execute()
-        .catch((error) => {
-          if (error.message.includes("statement timeout")) {
-            throw new NonRetryableError(error.message);
-          } else {
-            throw error;
+          const transactions: InternalTransaction[] = [];
+          const transactionReceipts: InternalTransactionReceipt[] = [];
+          const logs: InternalLog[] = [];
+          const traces: InternalTrace[] = [];
+
+          while (
+            transactionIndex < transactionsRows.length &&
+            transactionsRows[transactionIndex]!.block_number === block.number
+          ) {
+            const transaction = transactionsRows[transactionIndex]!;
+            transactions.push(decodeTransaction({ transaction }));
+            transactionIndex++;
           }
-        });
 
-      const events = rows.map((_row) => {
-        // Without this cast, the block_ and tx_ fields are all nullable
-        // which makes this very annoying. Should probably add a runtime check
-        // that those fields are indeed present before continuing here.
-        const row = _row as NonNull<(typeof rows)[number]>;
+          while (
+            transactionReceiptIndex < transactionReceiptsRows.length &&
+            transactionReceiptsRows[transactionReceiptIndex]!.block_number ===
+              block.number
```

Comment:
What is this check doing?

Reply:
This is how the "zipper" algorithm is implemented, gathering all event data for a given block